Sources/DuetBot/Core/Command.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Command.swift
+//  DuetBot
+//
+//  Created by ライアン on 11/10/18.
+//
+
+import Foundation
+import Sword
+
+// NOTE: If there are more ways to reply than just with a message, turn
+// reply into a real type and return that instead.
+typealias Reply = String
+
+struct Command {
+    
+    let pattern: String
+    let arguments: [String]
+    private let message: Message
+    
+    var sender: User? {
+        return message.author
+    }
+    
+    init?(_ message: Message) {
+        self.message = message
+        guard message.content.hasPrefix("!") else { return nil }
```

Comment:
why do this over !message.content.hasPrefix("!") { return nil}

Reply:
Mostly a preferential thing—sometimes one reads better than the other. Generally if you're going to use `!` you should replace it with guard unless it reads awkwardly. Also, you want whatever reduces nesting as much as possible.